Makefile

```diff
@@ -141,186 +111,63 @@ endif
-	mkdir -p ../mattermost-server/plugins
-	tar -C ../mattermost-server/plugins -zxvf dist/$(BUNDLE_NAME)
-else
-	@echo "No supported deployment method available. Install plugin manually."
-endif
+	./build/bin/deploy $(PLUGIN_ID) dist/$(BUNDLE_NAME)
+
+.PHONY: debug-deploy
+debug-deploy: debug-dist deploy
+
+.PHONY: debug-dist
+debug-dist: apply server webapp-debug bundle
 
 ## Runs any lints and unit tests defined for the server and webapp, if they exist.
 .PHONY: test
-test: .npminstall validate-go-version
+test: webapp/.npminstall
 ifneq ($(HAS_SERVER),)
-	$(GOTEST) -race -v ./server/...
+	$(GO) test -v $(GO_TEST_FLAGS) ./server/...
 endif
 ifneq ($(HAS_WEBAPP),)
 	cd webapp && $(NPM) run fix && $(NPM) run test;
 endif
 
 ## Creates a coverage report for the server code.
 .PHONY: coverage
-coverage: .npminstall validate-go-version
+coverage: webapp/.npminstall
 ifneq ($(HAS_SERVER),)
-	$(GOTEST) -race -coverprofile=server/coverage.txt ./server/...
-	$(GOTOOL) cover -html=server/coverage.txt
-endif
-ifneq ($(HAS_WEBAPP),)
-	cd webapp && $(NPM) run fix && $(NPM) run test-ci;
+	$(GO) test $(GO_TEST_FLAGS) -coverprofile=server/coverage.txt ./server/...
+	$(GO) tool cover -html=server/coverage.txt
 endif
 
 ## Extract strings for translation from the source code.
 .PHONY: i18n-extract
 i18n-extract:
 ifneq ($(HAS_WEBAPP),)
-	@[[ -d $(MM_UTILITIES_DIR) ]] || echo "You must clone github.com/mattermost/mattermost-utilities repo in .. to use this command"
-	@[[ -d $(MM_UTILITIES_DIR) ]] && cd $(MM_UTILITIES_DIR) && npm install && npm run babel && node mmjstool/build/index.js i18n extract-webapp --webapp-dir ../mattermost-plugin-demo/webapp
+ifeq ($(HAS_MM_UTILITIES),)
+	@echo "You must clone github.com/mattermost/mattermost-utilities repo in .. to use this command"
+else
+	cd $(MM_UTILITIES_DIR) && npm install && npm run babel && node mmjstool/build/index.js i18n extract-webapp --webapp-dir $(PWD)/webapp
+endif
 endif
 
 ## Clean removes all build artifacts.
 .PHONY: clean
 clean:
 	rm -fr dist/
 ifneq ($(HAS_SERVER),)
+	rm -fr server/coverage.txt
 	rm -fr server/dist
 endif
 ifneq ($(HAS_WEBAPP),)
-	rm -fr .npminstall
+	rm -fr webapp/.npminstall
+	rm -fr webapp/junit.xml
 	rm -fr webapp/dist
 	rm -fr webapp/node_modules
 endif
 	rm -fr build/bin/
 
-# Help documentatin à la https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
+# Help documentation à la https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
 help:
 	@cat Makefile | grep -v '\.PHONY' |  grep -v '\help:' | grep -B1 -E '^[a-zA-Z0-9_.-]+:.*' | sed -e "s/:.*//" | sed -e "s/^## //" |  grep -v '\-\-' | sed '1!G;h;$$!d' | awk 'NR%2{printf "\033[36m%-30s\033[0m",$$0;next;}1' | sort
-
-# sd is an easier-to-type alias for server-debug
-.PHONY: sd
-sd: server-debug
-
-# server-debug builds and deploys a debug version of the plugin for your architecture.
-# Then resets the plugin to pick up the changes.
-.PHONY: server-debug
-server-debug: server-debug-deploy reset
```

Comment:
@jfrerich Just to double check: Are we fine with removing these commands?

Reply:
@hanzei I did all the requested changes. Only missing clarification on this point. Should I revert this changes? Copied the makefile from autolink as indicated in the issue.

Reply:
I think we should keep and investigate these options.  They were implemented some time ago, but the idea being that we could save time by only building the appropriate architecture.

Reply:
Well, technically the are stored in Git and saved there. 1/5 that consistency is more important that the build features.